### PR TITLE
Migrate Puppeteer test action from Blocks repo

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Migrates [the workflow for running Puppeteer tests](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/.github/workflows/php-js-e2e-tests.yml) on PRs from the WooCommerce Blocks repo to the WooCommerce monorepo.

The workflow should ignore the other plugins, and only run on PRs in the Blocks plugin.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. 
2.
3.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/blocks-e2e.yml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+on:
+    pull_request:
+        paths-ignore:
+            - '**/changelog/**'
+            - '**/woo-ai/**'
+            - '**/woocommerce-admin/**'
+            - '**/woocommerce-beta-tester/**'
+            - '**/woocommerce-docs/**'
+            - '**/woocommerce/**'
```
Comment (ObliviousHarmony) [inline .github/workflows/blocks-e2e.yml]:
```suggestion
        paths:
            - 'plugins/woocommerce-blocks/**'
            - 'plugins/woocommerce/src/Blocks/**'
            - 'plugins/woocommerce/templates/**'
            - 'plugins/woocommerce/patterns/**'
```

Comment (ObliviousHarmony) [reply .github/workflows/blocks-e2e.yml]: In this case I think it's better to target the the paths associated with Blocks. Generally speaking, `paths` is preferable unless you _specifically_ want a PR to run on every change _except_ the paths you defined.

File: .github/workflows/blocks-e2e.yml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            - name: Set up PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: '8.0'
+                  coverage: none
+                  tools: composer
+
+            - name: Composer install
+              run: |
+                  composer install
```
Comment (ObliviousHarmony) [inline .github/workflows/blocks-e2e.yml]: I'm 99% sure that virtually _all_ of this can be replaced by using `setup-woocommerce-monorepo` 😄

**Review by ObliviousHarmony — COMMENTED** [review]
Just a few small comments. As an aside, something to think about is that these were originally run from their repository root. We should use [`working_directory`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory) to make sure that we're replicating that behavior.

**Comment by lanej0** [comment]
Thanks for those comments/tips @ObliviousHarmony!  I should have marked this as draft as it's still a work in progress :-)

**Review by ObliviousHarmony — APPROVED** [review]
It liiiiives

**Pull request merged**